scripts/ci/pre_commit/pre_commit_update_breeze_setup_hash.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import hashlib
+from pathlib import Path
+
+if __name__ not in ("__main__", "__mp_main__"):
+    raise SystemExit(
+        "This file is intended to be executed as an executable program. You cannot use it as a module."
+        f"To execute this script, run ./{__file__} [FILE] ..."
+    )
+
+AIRFLOW_SOURCES_ROOT = Path(__file__).parent.parent.parent.parent
+BREEZE_SOURCES_ROOT = AIRFLOW_SOURCES_ROOT / "dev" / "breeze"
+
+
+def get_package_setup_metadata_hash() -> str:
+    """
+    Retrieves hash of setup.py and setup.cfg files.
+
+    This is used in order to determine if we need to upgrade Breeze, because some
+    setup files changed. Blake2b algorithm will not be flagged by security checkers
+    as insecure algorithm (in Python 3.9 and above we can use `usedforsecurity=False`
+    to disable it, but for now it's better to use more secure algorithms.
+    """
+    the_hash = hashlib.new("blake2b")
+    the_hash.update((BREEZE_SOURCES_ROOT / "setup.py").read_bytes())
+    the_hash.update((BREEZE_SOURCES_ROOT / "setup.cfg").read_bytes())
```

Comment:
We should add `pyproject.toml` as well.

Reply:
Yeah. We don't have it yet - but I believe setuptools/pip are not yet ready to switch to pyproject.toml . Do you think it's worth to add it now ? what should we move there from cfg ? Or maybe we could already get rid of setup.py or setup.cfg ?

Reply:
Added it and also handled edge-case when Breeze is run on old version of sources (where ./dev/breeze/ is missin.

Reply:
Cool! Happy to review and learn :)